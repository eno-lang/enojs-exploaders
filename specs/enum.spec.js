const eno = require('enojs');
const { enumFactory } = require('../loaders.js');

const examples = {
  'eno'  : 'eno',
  'json' : 'json',
  'yaml' : 'yaml',
  'YAML' : false,
  'car'  : false,
  '13'   : false
};

const language = enumFactory(['eno', 'json', 'yaml']);

describe('enum loader', () => {
  for(let [value, result] of Object.entries(examples)) {
    describe(value, () => {
      const document = eno.parse(`value: ${value}`);

      if(result === false) {
        it('throws an error', () => {
          expect(() => document.field('value', language)).toThrowErrorMatchingSnapshot();
        });
      } else {
        it(`returns ${result}`, () => {
          expect(document.field('value', language)).toBe(result);
        });
      }
    });
  }
});
