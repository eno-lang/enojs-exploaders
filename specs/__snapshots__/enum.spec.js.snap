// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enum loader 13 throws an error 1`] = `
"'value' must be one of 'eno', 'json', 'yaml'.

   Line | Content
 >    1 | value: 13
"
`;

exports[`enum loader YAML throws an error 1`] = `
"'value' must be one of 'eno', 'json', 'yaml'.

   Line | Content
 >    1 | value: YAML
"
`;

exports[`enum loader car throws an error 1`] = `
"'value' must be one of 'eno', 'json', 'yaml'.

   Line | Content
 >    1 | value: car
"
`;
